### DISCUSSION TITLE
Adventure Map, Oracle screen, right click on the heroes' portraits displays information about the heroes

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10297] [General]
In the Oracle screen, it would be nice to be able to click on the portraits of the heroes to get the information about the heroes.

In fheroes2:

![Oracle 04](https://user-images.githubusercontent.com/43583994/92477396-d19e4400-f1e0-11ea-86b4-bc8533d0d088.png)

**Comment by ihhub** [comment]
I think it'll be cheating as for example you haven't even met a hero but you will approximately know his army.

**Comment by Branikolog** [comment]
@ihhub only for the cases when hero is in the visible part of a world map, I may suggest...

**Comment by LeHerosInconnu** [comment]
Hello @ihhub and @Branikolog,

> I think it'll be cheating as for example you haven't even met a hero but you will approximately know his army.

> @ihhub only for the cases when hero is in the visible part of a world map, I may suggest...

I was about to mention that. :)
For the player's hero, the full information window is displayed.
For the opponent heroes visible on the adventure map (outside the Fog of War), the standard info window is displayed.
For the opponent heroes not visible on the adventure map (under the Fog of War), the troops are replaced by question marks.
If the "Identify Hero" spell is active and/or the artifact "Crystall Ball" is effective, for the opposing heroes visible on the adventure map (outside the Fog of War), the full information window is displayed.

**Comment by Branikolog** [comment]
> For the opponent heroes not visible on the adventure map (under the Fog of War), the troops are replaced by question marks.

For this case what additional info can we show in a popup window? Portrait and hero stats are already visible in oracle main window... Or is it just to keep the behaviour of popping up window on a click?

edit: Hello, @LeHerosInconnu! :)

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog, :D

> > For the opponent heroes not visible on the adventure map (under the Fog of War), the troops are replaced by question marks.
> 
> For this case what additional info can we show in a popup window? Portrait and hero stats are already visible in oracle main window... Or is it just to keep the behaviour of popping up window on a click?
> 
> edit: Hello, @LeHerosInconnu! :)

The information displayed depends on the number of Thieves' Guilds built.
In case, there are enough Thieves' Guilds built to see the hero's primary skill values (or in the Oracle screen), the info window should display the same information.

**Comment by ihhub** [comment]
Moving this to the Discussion section as we have disagreements on this idea.
